aw-awatcher: init at 0.2.7

BhasherBEL opened the pull request:
## Description of changes
https://github.com/2e3s/awatcher

 Activity and idle watchers for activitywatch compatible with wayland and hyprland.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

BhasherBEL committed:
aw-awatcher: init at 0.2.7

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4066

0x61nas commented on pkgs/by-name/aw/aw-awatcher/package.nix:
```diff
@@ -0,0 +1,38 @@
+
+  buildInputs = [ openssl ];
+
+  meta = with lib; {
```
could you avoid using `with lib` here because it is a problematic anti-pattern, refer to: https://github.com/NixOS/nixpkgs/issues/208242

0x61nas commented on pkgs/by-name/aw/aw-awatcher/Cargo.lock:
the upstream have [`Cargo.lock`](https://github.com/2e3s/awatcher/blob/main/Cargo.lock) already, why do we need to maintain a our own one?

BhasherBEL replied:
Because based on [the documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md) and my experiments, it's not working as there is Cargo `git` dependencies.

0x61nas replied:
you can achieve the same thing by:

```nix
  cargoLock = {
    lockFile = src + /Cargo.lock;
    outputHashes = {
      "aw-client-rust-0.1.0" = "sha256-fCjVfmjrwMSa8MFgnC8n5jPzdaqSmNNdMRaYHNbs8Bo=";
    };
  };
```

by doing this you'll use `Cargo.lock` from the upstream source and you can provide the git deps hashes. refer to: https://github.com/NixOS/nixpkgs/pull/316252/files#diff-8ac422c1d925671c1049e51f44d5fe3338fce5e9e1dce3fa0e7d1f6e3cfe1425R28

0x61nas commented on pkgs/by-name/aw/aw-awatcher/package.nix:
```diff
@@ -0,0 +1,38 @@
+    description = "Activity and idle watchers";
+    homepage = "https://github.com/2e3s/awatcher";
+    license = lib.licenses.mpl20;
+    maintainers = [ lib.maintainers.bhasher ];
```
```suggestion
    maintainers = with lib.maintainers; [ bhasher ];
```

0x61nas commented on pkgs/by-name/aw/aw-awatcher/package.nix:
```diff
@@ -0,0 +1,38 @@
+    license = lib.licenses.mpl20;
+    maintainers = with lib.maintainers; [ bhasher ];
+    mainProgram = "awatcher";
+    platforms = lib.platforms.linux;
```
am not sure if we need to mark it as broken on `darwin` if we already exclude it here. but I suppose it's not required by this.

0x61nas reviewed (approved):
works for me, thanks

sifmelcara commented:
This is duplicate of #287858 (which is stalled), also here is a reference to the packaging request: #287826

@aikooo7 are you still interested in merging the PR?

aikooo7 commented:
> This is duplicate of #287858 (which is stalled), also here is a reference to the packaging request: #287826
> 
> @aikooo7 are you still interested in merging the PR?

My bad for being so much time out first of all.

I could fix it and merge it but since it is done here and people say it works we should perhaps merge this one?

aikooo7 commented on maintainers/maintainer-list.nix:
Shouldn't this be in a separate commit?

BhasherBEL replied:
I can split it if it's necessary

aikooo7 commented:
> This is duplicate of #287858 (which is stalled), also here is a reference to the packaging request: #287826
> 
> @aikooo7 are you still interested in merging the PR?

I had indeed some stuff to do in that pr that I didn't see, my bad. Everything should be fixed now and now I am also using the latest version

BhasherBEL commented:
Feel free to merge the other PR, but it seems stale and is several versions behind.

aikooo7 commented:
> Feel free to merge the other PR, but it seems stale and was several versions behind. It also include the `Cargo.lock`, by the opposite to here.

1. It is in the same version as yours, wdym?
2. By my research it is kinda standard and happens In a lot of package to have the Cargo.lock, what is the issue?

BhasherBEL commented:
>     1. Thanks for callout, I did destroy the commit that I had changed to use the latest version and didn't notice it, it should be the latest version now.

Yup, it's perfect now, my comment was anterior to that change.

>     2. You pull the Cargo.lock at build time, I pre-pulled it, both might exist and I am using what I saw in the rust's nixos documentation, what is the issue?

I can agree with you and did that at first, but I changed due to a [comment](https://github.com/NixOS/nixpkgs/pull/315864#discussion_r1633257386). 

To be honest I don't really care which PR is merged, but I would love to see this package in `nixpkgs`, so if it's faster with your PR, I'm in.

BhasherBEL commented:
Closed in favor of https://github.com/NixOS/nixpkgs/pull/287858, which is now more up to date.
